src/Workspaces/CSharp/Portable/Formatting/Rules/SuppressFormattingRule.cs

```diff
@@ -43,6 +43,24 @@ private void AddSpecificNodesSuppressOperations(List<SuppressOperation> list, Sy
                 return;
             }
 
+            if (node is DeconstructionPatternSyntax deconstructionPattern)
+            {
+                AddSuppressWrappingIfOnSingleLineOperation(list, deconstructionPattern.OpenParenToken, deconstructionPattern.CloseParenToken);
+                return;
+            }
+
+            if (node is PropertySubpatternSyntax propertySubpattern)
+            {
+                AddSuppressWrappingIfOnSingleLineOperation(list, propertySubpattern.OpenBraceToken, propertySubpattern.CloseBraceToken);
+                return;
+            }
+
+            if (node is SwitchExpressionArmSyntax switchExpressionArm)
+            {
+                AddSuppressWrappingIfOnSingleLineOperation(list, switchExpressionArm.GetFirstToken(), switchExpressionArm.GetLastToken());
+                return;
+            }
```

Comment:
do you need a similar suppressoin for the entire switch expression?

Reply:
Done. Thanks